website/documentation/050-tutorials/content/howto/add-node-to-cluster/_index.md

```diff
@@ -0,0 +1,80 @@
+type: tutorial-page
+level: advanced
+index: 10
+reviewer: Gaurav Gupta
+status: 
+last_reviewed: 
+category: Operation
+scope: app-developer
+---
+
+## Term clarification
+Here we will look at the steps on how to add a node to an existing cluster without the support of Gardener. Such a node will not be managed by Gardener, and if it goes down for any reason, Gardener will not be responsible to replace it.
+
+## Why
+This tutorial is useful for those end-users who need specifically configured nodes, which are not yet supported by Gardener. For example: an end-user who wants some workload that require runnc instead of runc as container runtime.
+
+## How
+1. Create a new instance in the same VPC/network as other machines in the cluster. You should be able to ssh into the machine. So save its private key, and assign public IP to it. If adding public IP is not preferred, then ssh into any other machine in the cluster, and then ssh into the new machine using its private key.
+
+   To ssh into a machine which is already in the cluster, use the steps defined [here](https://github.wdf.sap.corp/pages/kubernetes/gardener/015-tutorials/content/howto/ssh-into-node/ "ssh-into-node").
+
+   Also, while creating the new machine, attach same IAM role to it as that attached to the existing machines in the cluster. This is required by kubelet in the new machine so that it can contact the cloud provider to query the node's name.
+
+1. In the new machine, create file `/var/lib/kubelet/kubeconfig-bootstrap` with the following content:
+
+```yaml
+apiVersion: v1
+kind: Config
+current-context: kubelet-bootstrap@default
+clusters:
+- cluster:
+    certificate-authority-data: $CA_CRT
+    server: $SERVER
+  name: default
+contexts:
+- context:
+    cluster: default
+    user: kubelet-bootstrap
+  name: kubelet-bootstrap@default
+users:
+- name: kubelet-bootstrap
+  user:
+    as-user-extra: {}
+    token: $TOKEN
+```
+
+3. ssh into an existing node, and run these commands to get the values of $CA_CRT and $SERVER to be replaced in above file:
+- $SERVER
+```bash
+ /opt/bin/hyperkube kubectl --kubeconfig /var/lib/kubelet/kubeconfig-real config view -o go-template='{{index .clusters 0 "cluster" "server"}}' --raw
+ ```
+- $CA_CRT
+```bash
+/opt/bin/hyperkube kubectl --kubeconfig /var/lib/kubelet/kubeconfig-real config view -o go-template='{{index .clusters 0 "cluster" "certificate-authority-data"}}' --raw
+```
+
+4. $TOKEN\
+  A bootstrap token needs to be constructed which can be used by kubelet in the new machine to add itself to the cluster. Kube-apiserver uses a secret in the `kube-system` namespace to authenticate this token. This token is valid for 90 minutes from the time of creation, and the corresponding secret captures this detail in its `.data.expiration` field. The name of this secret is of the format `bootstrap-token-*`. Gardener takes care of creating new bootstrap tokens, and creating corresponding secrets.
```

Comment:
The kubelet on the new machine needs a bootstrap token to authenticate with the kube-apiserver when adding itself to the cluster. ...
Gardener takes care of creating new bootstrap tokens and the corresponding secrets.